datafusion/sql/src/planner.rs

```diff
@@ -3165,6 +3171,64 @@ fn wrap_projection_for_join_if_necessary(
     Ok((plan, need_project))
 }
 
+/// Ensure any column reference of the expression is determined.
+/// Assume we have two schema:
+/// schema1: a, b ,c
+/// schema2: a, d, e
+///
+/// `schema1.a + schema2.a` is determined.
+/// `a + d` is not determined, because `a` may come from schema1 or schema2.
+fn ensure_any_column_reference_is_determined(
+    expr: &Expr,
+    schemas: &[DFSchemaRef],
+) -> Result<()> {
+    if schemas.len() == 1 {
+        return Ok(());
+    }
+
+    // extract columns both in one more schemas.
+    let mut column_count_map: HashMap<String, usize> = HashMap::new();
```

Comment:
You might be able to avoid a clone into a string here. Maybe a follow on PR

```suggestion
    let mut column_count_map: HashMap<&str, usize> = HashMap::new();
```